### PR TITLE
905033-Added UG documentation for newly implemented Mathematical functions

### PR DESCRIPTION
**Pull request by ushasundarajan**
Added UG documentation of newly implemented mathematical formulas in Calculate that are listed below:

1.  XLookUp
2.  XMatch
3.  FloorMath
4.  FloorPrecise
5.  IsoCeliing
6.  CeilingPrecise
7.  HStack
8.  VStack
9.  Phi

### CODE REVIEW

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 1]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+
+* Using [match_mode] allows you to control whether the function looks for exact matches, approximate matches, or supports wildcard characters.   
+
+
+
+* The [search_mode] argument lets you control the order in which the search is performed, which can be useful for optimizing performance on large datasets.   
+
+
+
+## XMATCH
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2647]: @ushasundarajan 

The XMATCH function returns the relative position of an item in a range, similar to MATCH, but with additional capabilities for exact and approximate matches.

Syntax:

XMATCH(lookup_value, lookup_array, [match_mode], [search_mode])

Where:

lookup_value: The value to search for.
lookup_array: The range or array to search.
[match_mode] (Optional): Specifies the match type:
0: Exact match (default).
-1: Exact match or next smallest item.
1: Exact match or next largest item.
2: Wildcard match using *, ?, and ~.
[search_mode] (Optional): Specifies the search order:
1: Search from first to last (default).
-1: Search from last to first.
2: Binary search in ascending order (data must be sorted).
-2: Binary search in descending order (data must be sorted).

Remarks:

XMATCH is useful for retrieving the position of a value within a range, with more flexibility than MATCH.
It supports both normal and reverse search directions, as well as wildcard matching.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 2]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+
+    If not sorted, invalid results will be returned.   
+
+
+
+  * -2 - Perform a binary search that relies on lookup_array being sorted in descending order. If not sorted, invalid results will be returned.   
+
+
+
+## FLOOR.MATH
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2719]: @ushasundarajan 

The FLOOR.MATH function rounds a number down to the nearest integer or a specified multiple of significance.

Syntax:

FLOOR.MATH(number, [significance], [mode])

Where:

number: The number to be rounded down.
significance (Optional): The multiple to which to round down. Defaults to 1.
mode (Optional): Controls rounding of negative numbers. If omitted, negative numbers are rounded away from zero.
Remarks:

By default, positive numbers are rounded down to the nearest integer (e.g., 6.3 becomes 6).
Negative numbers are rounded away from zero unless the mode argument is used (e.g., -6.7 becomes -7, but with mode set to -1, -6.3 becomes -6).
The significance argument defines the multiple to which the number is rounded down. For example, rounding 7.5 with a significance of 3 results in 6.
If number divided by significance has a remainder, the result is rounded down.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 3]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+* The significance argument rounds the number down to the nearest integer that is a multiple of the significance specified. The exception is where the number to be rounded is an integer. For example, for a significance of 3, the number is rounded down to the next integer that is a multiple of 3.
+
+
+
+* If number divided by a significance of 2 or greater results in a remainder, the result is rounded down.
+
+
+
+## FLOOR.PRECISE
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2776]: @ushasundarajan 
The FLOOR.PRECISE function rounds a number down to the nearest integer or a specified multiple of significance, regardless of the sign.

Syntax:

FLOOR.PRECISE(number, [significance])

Where:

number: The number to be rounded down.
significance (Optional): The multiple to which the number should be rounded. Defaults to 1.

Remarks:

The absolute value of significance is used, meaning FLOOR.PRECISE always rounds down, regardless of whether the number is positive or negative.
If either the number or significance is zero, the function returns zero.
The rounding behavior remains consistent regardless of the signs of the number or significance.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 4]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+* If an array has fewer columns than the maximum width of the selected array.
+
+
+
+* `VSTACK` returns a `#N/A` error in the additional columns. Use `VSTACK` inside the `IFERROR` function to replace `#N/A` with the value of your choice.
+
+
+
+## PHI
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2976]: @ushasundarajan 
The PHI function returns the value of the probability density function for the standard normal distribution for a given number.

Syntax:

PHI(x)

Where:

x: The number for which the probability density is calculated.
Remarks:

If x is not a valid numeric value, PHI will return the #NUM! error.
If x is of an invalid data type, PHI will return the #VALUE! error.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 5]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+* If an array has fewer rows than the maximum width of the selected arrays.
+
+
+
+* `HSTACK` returns a `#N/A` error in the additional rows. Use `HSTACK` inside the `IFERROR` function to replace `#N/A` with the value of your choice.
+
+
+
+## VSTACK
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2928]: @ushasundarajan 

The VSTACK function appends arrays vertically (one below the other) to return a larger array.

Syntax:

VSTACK(array1, [array2], ...)

Where:

array: The arrays to be appended vertically.
Remarks:

VSTACK returns an array where each input array is appended in a row-wise fashion.
Rows: The result will have the combined number of rows from all input arrays.
Columns: The result will have the maximum column count of the input arrays.
If an array has fewer columns than the maximum, the additional columns will contain #N/A. You can use IFERROR to replace these errors with a specified value.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 6]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+## HSTACK
+
+
+
+The `HSTACK` function appends arrays horizontally and in sequence to return a larger array.
+
+
+
+**Syntax:**
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2889]: @ushasundarajan 
The HSTACK function combines arrays horizontally (side by side) to form a larger array.

Syntax:

HSTACK(array1, [array2], ...)

Where:

array: The arrays to be appended horizontally.

Remarks:

HSTACK returns an array where each input array is appended in a column-wise fashion.
Rows: The number of rows in the result will be the maximum number of rows across all input arrays.
Columns: The number of columns will be the sum of the columns from all input arrays.
If an array has fewer rows than the maximum row count, #N/A will fill the missing rows. You can use the IFERROR function to handle these errors.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 7]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+
+
+## CEILING.PRECISE
+
+
+The function `CEILING.PRECISE` returns a number that is rounded up to the nearest integer or to the nearest multiple of significance. Regardless of the sign of the number, the number is rounded up. However, if the number or the significance is zero, zero is returned.
+
+
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2852]: @ushasundarajan 

The CEILING.PRECISE function rounds a number up to the nearest integer or a specified multiple of significance. Like ISO.CEILING, it always rounds up regardless of whether the number is positive or negative. If the number or significance is zero, it returns zero.

Syntax:

CEILING.PRECISE(number, [significance])

Where:

number: The value to be rounded up.
significance (Optional): The multiple to which the number is rounded. If omitted, it defaults to 1.

Remarks:

CEILING.PRECISE uses the absolute value of the multiple, ensuring that the function rounds up regardless of the signs of the number or significance.
If the number or significance is zero, the result is zero.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 8]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
...
+The `ISO.CEILING` function returns a number that is rounded up to the nearest integer or to the nearest multiple of significance. Regardless of the sign of the number, the number is rounded up. However, if the number or the significance is zero, zero is returned.
+
+
+
+**Syntax:**
+
+
+
+_ISO.CEILING(number, [significance])_
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2822]: @ushasundarajan 
The ISO.CEILING function rounds a number up to the nearest integer or a specified multiple of significance. It always rounds up, regardless of whether the number is positive or negative. If the number or significance is zero, it returns zero.

Syntax:

ISO.CEILING(number, [significance])

Where:

number: The value to be rounded up.
significance (Optional): The multiple to which the number is to be rounded. If omitted, it defaults to 1.

Remarks:

The absolute value of the multiple is used, meaning ISO.CEILING always rounds up, regardless of the sign of the number or significance.
If either number or significance is zero, the function returns zero.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md [hunk 9]
```diff
@@ -2533,3 +2533,475 @@ _SERIESSUM(x, n, m, coefficients)_
 
 
 * The argument is non-numeric, it returns the `#VALUE!` error message.
+
+
+
+## XLOOKUP
+
```
[the remark this answers is not in the document]
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2540]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 1 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2643]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 2 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2722]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 8 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2828]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 7 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2862]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 6 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2903]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 5 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2942]: @Manivannan-E Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md (hunk 4 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Math-Trigonometry.md:2990]: @Manivannan-E Changes updated.

**Review by Manivannan-E — APPROVED** [review]
Changes are fine.

**Pull request merged**